packages/dev/core/src/DeviceInput/gestureRecognizer.ts

```diff
@@ -0,0 +1,46 @@
+import { PointerEventTypes } from "../Events/pointerEvents";
+import type { IPointerEvent, IWheelEvent } from "../Events/deviceInputEvents";
+import type { DeviceType } from "./InputDevices/deviceEnums";
+import { PointerInput } from "./InputDevices/deviceEnums";
+import type { DeviceSource } from "./InputDevices/deviceSource";
+import { Scene } from "../scene";
+
+/** @internal */
+export class GestureRecognizer {
+    private static _PREVIOUS_POINTER_TIME: number = 0;
+    private static _DOUBLE_CLICK_OCCURED: boolean = false;
+
+    /** @internal */
+    public static DeterminePointerEventType(deviceSource: DeviceSource<DeviceType.Mouse | DeviceType.Touch>, eventData: IPointerEvent | IWheelEvent) {
+        const doubleClickDelay = Scene.DoubleClickDelay;
+        const wheelEventData = eventData as IWheelEvent;
+        const pointerEventData = eventData as IPointerEvent;
+        let type = 0;
+
+        if (wheelEventData.deltaMode !== undefined) {
+            type = PointerEventTypes.POINTERWHEEL;
+        } else if (pointerEventData.inputIndex === PointerInput.Move) {
+            type = PointerEventTypes.POINTERMOVE;
+        } else if (deviceSource.getInput(pointerEventData.inputIndex) === 1) {
+            type = PointerEventTypes.POINTERDOWN;
+        } else {
+            type = PointerEventTypes.POINTERUP;
+
+            if (Date.now() - this._PREVIOUS_POINTER_TIME < doubleClickDelay && !this._DOUBLE_CLICK_OCCURED && type === PointerEventTypes.POINTERUP) {
+                this._DOUBLE_CLICK_OCCURED = true;
+                console.log("Double click occured");
+                type += PointerEventTypes.POINTERDOUBLETAP;

```

Comment:
This technically works I guess, but it's a bit strange to use for bitwise operations. Why not this?
```suggestion
                type &= PointerEventTypes.POINTERDOUBLETAP;
```